### PR TITLE
SCP 294

### PR DESCRIPTION
**Pull request by Zth--**
I'll open this for discussion, this adds scp 294 (read the link for more info)
http://www.scp-wiki.net/scp-294
http://puu.sh/n8nJi/72aac20bb1.png

this machine is OP since it can dispense stuff like spirit breaker toxin, so the question is:
should this be an admin-only spawnable machine or should it be edited to be a large artifact?

### CODE REVIEW

**Comment by Zth--** [comment]
4 commits? Fuck I think I broke something

**Comment by 9600bauds** [comment]
Put
It
In
Centcomm

**Comment by Shadowmech88** [comment]
> this machine is OP since it can dispense stuff like spirit breaker toxin

You could always add a list of reagents that are prohibited.

**Comment by 9600bauds** [comment]
That's against the entire point of the premise

**Comment by Shadowmech88** [comment]
Better to have a gimped version that can be allowed to enter rounds rather than a version so OP that it just sits in Centcomm never being used.

If you really wanted to, you could even make a child that has the list nulled out, specifically for adminbus.

**Comment by Zth--** [comment]
That's why I'm asking for feedback. I'll add a different version of the machine that displays "OUT OF RANGE" whenever someone tries to get prohibited reagents.

Maybe the OP version it's just too OP for events

**Comment by ghost** [comment]
It doesn't even have an exception for adminodrazine and you're worried about spirit breaker?

**Comment by Zth--** [comment]
I couldn't remember that reagents name, thanks!

**Comment by 9600bauds** [comment]
I'm not very well-versed in the adminbus, but in what situation would it be OK to give someone infinite doctor's delight, creatine, cyanide, nitroglycerin, peridaxon, amutationtoxin etc. but not spiritbreaker?

**Comment by Shadowmech88** [comment]
Spiritbreaker's a traitor item only, isn't it?
All that other stuff can be obtained via chemistry/botany normally, this machine would just expedite the process.

**Comment by Zth--** [comment]
Ok I added an exception for adminodrazine, since it's an admin-only reagent. It still can be VVed though

**Comment by LordMyroc** [comment]
I don't really like the inclusion of SCPs in this game. I mean, it looks nice and all, and would probably make for a nice xenoarcheology artifact, but don't explicitly call it an SCP.

**Comment by PJB3005** [comment]
Wait a second why do you need to insert a beaker or anything into it, it makes cups itself!

File: code/game/machinery/scp_294.dm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		// the ui does not exist, so we'll create a new() one
+        // for a list of parameters and their descriptions see the code docs in \code\\modules\nano\nanoui.dm
+		ui = new(user, src, ui_key, "scp_294.tmpl", "[src.name]", 390, 315)
+		// when the ui is first opened this is the data it will use
+		ui.set_initial_data(data)
+		// open the new ui window
+		ui.open()
+
+/obj/machinery/chem_dispenser/scp_294/Topic(href, href_list)
+	if(href_list["close"])
```
Comment (PJB3005) [inline code/game/machinery/scp_294.dm]: Remove, this is 100% oldcoders.

File: code/game/machinery/scp_294.dm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		if(usr.machine == src) usr.unset_machine()
+		return 1
+	if(stat & (NOPOWER|BROKEN))
```
Comment (PJB3005) [inline code/game/machinery/scp_294.dm]: Sanity is more than this, specifically `..()`

File: code/game/machinery/scp_294.dm [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+//										//
+//	This is a child of a chemistry		//
+//	dispenser. Info of how it works at	//
+//	http://www.scp-wiki.net/scp-294		//
+//										//
+//////////////////////////////////////////
+
+/obj/machinery/chem_dispenser/scp_294
+	name = "\improper SCP 294"
+	desc = "It appears to be a standard coffee vending machine, the only noticeable difference being an entry touchpad with buttons corresponding to an English QWERTY keyboard."
```
Comment (ghost) [inline code/game/machinery/scp_294.dm]: It's not English it's common

Comment (gbasood) [inline code/game/machinery/scp_294.dm]: Galactic Common****
Alternatively just leave out the English part.

**Comment by Zth--** [comment]
I did some further testing, things like emps and smokes cannot be created.
Now the last debate, should this be called "strange coffee machine" or should it remain as scp 294?
Add this to xenoarch, Y/N? (I have no idea how, though)

**Comment by SysadminJeroen** [comment]
In my opinion, it should be called SCP 294 when spawned by Admins.
And this should be added to Xenoarch.

**Comment by 9600bauds** [comment]
Large Xenoarch artifact? Eh, sure, I think spiritbreaker is not a big deal anyways compared to other stuff you can get.
Plus, I am of the opinion that it should either have no restrictions or not be in. No adminordrazine is fine though.

File: code/game/machinery/scp_294.dm [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+/obj/machinery/chem_dispenser/scp_294
+	name = "SCP 294"
+	desc = "It appears to be a standard coffee vending machine, the only noticeable difference being an entry touchpad with buttons corresponding to a Galactic Common QWERTY keyboard."
+	icon = 	'icons/obj/vending.dmi'
+	icon_state = "coffee"
+	energy = 10
+	max_energy = 10
+	amount = 10
+	dispensable_reagents = null
+	var/list/prohibited_reagents = list("adminodrazine")
```
Comment (ghost) [inline code/game/machinery/scp_294.dm]: double check that actually is the name of the chemical

Comment (Zth--) [inline code/game/machinery/scp_294.dm]: It's missing a "r"

**Comment by ghost** [comment]
Can use http://containmentbreach.wikia.com/wiki/SCP-294 as a reference too, it's an example of SCP-294 as applied in an actual game

**Comment by icantthinkofanameritenow** [comment]
I agree with Lord Myroc. We don't need references from a has-been series of shitty creepypastas. Nor should we add it to Xenoarch. 

Really, so many things you could make and THIS is the best you can do?

**Comment by gbasood** [comment]
@icantthinkofanameritenow I'm sorry your snowflake race idea didn't get added but that doesn't mean you should shit on other people's smaller-scale bussable ideas.

**Comment by Boxe** [comment]
@gbasood the hell is your problem

**Comment by ghost** [comment]
savage

**Comment by 9600bauds** [comment]
burn so sick wwjnc woke up from the grave

**Comment by icantthinkofanameritenow** [comment]
@gbasood
And I'm sorry you don't see how this item is just a license to griff to hell and back. Think about it- it can produce an infinite supply of cyanide, mutagen, space lube, and more. Do you really want that bussed in on a round where you expect the station to not be shit instantly? 

At least don't use the SCP name. We don't need any more references than we already have at any given time, I'm sick of them.

**Comment by 9600bauds** [comment]
Large artifacts are already almost always a griff-exclusive tool, aren't they?
By the way, chemdispensers can already make two of the three things you described, and cyanide can be easily substituted by chloral+mutagen. So essentially all three.

**Comment by gbasood** [comment]
Just to be clear, I don't have a problem with anyone. But seeing someone say 'man why would you code this when you could code BETTER THINGS' when they don't like a feature someone actually went ahead and wrote up the code for is literally the scummiest thing you could do.

@icantthinkofanameritenow I really don't see any problem with a SCP reference, especially considering you'll only see that reference on the rare occasion it gets bussed. If someone griffs with it, it's the admin's fault for spawning it in in the first place.

**Comment by icantthinkofanameritenow** [comment]
@9600bauds

No, they're more likely to kill the Xenoarch who found them more than anything else. And chemdispensers actually require intermediate steps and need recharging time. This has neither as far as I can see. 

And again, calling it an SCP is shit. If you absolutely must add it, call it an omnidispenser or something so it sounds like it would actually belong in the game.

**Comment by 9600bauds** [comment]
Seems to me like this does require charge time, doesn't it?
Intermediate steps mean basically fucking nothing other than the memorization/wiki factor by the way, this is coming from a man who has done chemistry a hundred times over

**Comment by SysadminJeroen** [comment]
@icantthinkofanameritenow Let's remove Xenoarcheology then, nearly everything you find can be abused in Xenoarch.
Let's remove Botany as well, you can produce so much (edit: **deadly**) crap in it.

Oh wait, I forgot. Let's remove Virology. Because it is so easy to make a somewhat deadly virus.

Maybe remove Chemistry? You can kill someone within a minute with bad chems.

We could remove Atmos, as they can do anything with distro.

See what I'm getting at? There are a ton of ways people can griff, it can also be made not OP by having a cooldown after usage.

**Comment by icantthinkofanameritenow** [comment]
@gbasood

The issue I have is that the thing it's referencing is terrible, and we have enough gratuitous references already. Especially among the adminbus items. 

You'll have to pardon my frustration, but unless you can direct me to "Coding for Retards, SS13 edition" all I can do is make comments and play an extremely bitter devil's advocate.

**Comment by 9600bauds** [comment]
Idunno, I like the thing it's referencing

**Comment by 9600bauds** [comment]
@icantthinkofanameritenow
http://game.ss13.moe/wiki/index.php?title=Guide_to_Contributing_to_the_Game
This is in the front page of the wiki

**Comment by gbasood** [comment]
@icantthinkofanameritenow I don't think most of our playerbase would hate the reference so much that they'd be willing to drop the feature over the name of the thing. I really do think that's just you.

I do take issue with the amount of times I've seen SCP-173 this week though, don't get me wrong, but that has nothing to do with it being an SCP reference and more to do with annoying overbus.

**Comment by icantthinkofanameritenow** [comment]
@9600bauds
Fine, I guess I'll start with the simple task of tweaking the plasma weapons so they're actually worth taking by making them deal more damage. Expect it to be done when it's done. 

@gbasood

Trust me,  "SCP" has become synonymous with "superpowered Mary Sue", and I say this as someone who used to follow it avidly. Besides, the name " omnidispenser" will make it sound like it actually belongs in SS13.

**Comment by Shadowmech88** [comment]
"References are good" vs "References are bad" has long been a point of contention on /vg/station. Because it's ultimately a point of personal taste, neither side can successfully convince the other.

Which means that the smaller side is unfortunately just going to have to suck it up. The goal here is for a majority of players to have fun with the things that are added. If you don't think something's fun, and you're in the minority? Tough luck.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88

You can hardly say that when you can't even tell which side is the minority. There's 6 people on this at most in a server that has at least 11 times that many people on it.

**Comment by 9600bauds** [comment]
Some feedback here: This does require charge, however it's not shown anywhere what the charge is. There is no indication that the machine is out of charge, it will just not do anything.
It takes a beaker, which in my opinion isn't very good. I feel like it would give it a lot more personality if it dispensed paper cups, like the thing it's referencing.

**Comment by Shadowmech88** [comment]
@icantthinkofanameritenow 
I can say that one side is a minority based on currently available evidence.
If you believe there are a ton of players that would hate this name, yell at them in the thread to make a Github account and voice their opinions here. It's silly to base decisions on how large a certain side could hypothetically be, rather than how large a certain side is shown to be by number of Github posters.

Or you could even ask Pomf to make a "References Good vs References Bad" server poll.

**Comment by ghost** [comment]
Not a big fan of calling it an SCP. Even the other SCP which is absolutely awful (sorry @dylanstrategie) calls itself "creature" in the code and otherwise.

**Comment by NigglyWiggly** [comment]
Calling any SCP and SCP is silly because the people on the station have no way of knowing what an SCP even is.

Also, just add a cooldown to the machine and a whitelist to what chemicals it can produce. That way you can't just have 20 buckets of Adminordrazine right away.
That, or make it a thing that's only in centcomm and just wait for some adming (Celt) to spawn it on the station during a round and ruin everything

Also @DrCelt I'm pretty sure 173 is referred to as 'Sculpture' or something in the code

**Comment by ghost** [comment]
Creature, sculpture and difference you know what I meant

**Comment by gbasood** [comment]
Based on that argument, we should probably call it "strange coffee machine" or something similar.

**Comment by Zth--** [comment]
I coded this solely for the purpose of learning. It's something easy and fun to code and this was a good example to get into templates. I have no strong feelings about this artifact, some people were requesting it and I thought it was something fun to do.

Yes, it has energy, (1/10 of a chem dispenser), and we agree that it should be renamed to "strange coffee machine". There's a good reason to add it to xenoarch: this machine will appear once every month or even more rare, so yes, having such a uniqe artifact should be rewarding.

Lastly, hey @icantthinkofanameritenow, being such a hideous faggot is annoying for us, thanks for your feedback but fuck you for the shitposting, get your shit on 4chan or somewhere else.

**Comment by ghost** [comment]
@Zth-- :+1:

**Comment by gbasood** [comment]
I don't agree with the SCP hate but I do agree that it should have some sort of stupid mysterious name
You just dug up a thing from the dirt that looks like a coffee machine, and your first reaction is "This thing's name is SCP-294 DOES seem silly to me.

I think stupid mysterious name would add more flavor to it than SCP-294

**Comment by clusterfack** [comment]
Yeah no explicit reference to SCP please.

**Comment by Zth--** [comment]
I was sure I pushed the commit a few days ago

**Comment by 9600bauds** [comment]
A little sad that it doesn't dispense paper cups

**Comment by Zth--** [comment]
I'll probably change it later, I have little time to work with this right now.

**Pull request merged**